### DISCUSSION TITLE
Pass arrays filter

### DISCUSSION THREAD

**Discussion by fgallardo-mpie** [pyvista/pyvista#2097] [Q&A] [answered]
Hello, 

I want to remove an array from my dataset. Something like

PolyDataFilters.remove_array('NameOfTheArray')

Just like Pass arrays work in Paraview (https://discourse.paraview.org/t/remove-attribute/3108).

I have only seen DataSet.clear_data() which removes all of them but doesn´t allow selection. 

Also, it would be nice, when saving a file, to choose which array to save just like in Paraview. This would also solve my current problem. 

Is any of these features already in pyvista? I have been looking for a while and couldn´t find them.

Many thanks in advance,
Francisco José Gallardo Basile

**Comment by akaszynski** [comment, marked as answer]
Hello,
You can and remove point or cell arrays as if it was a dictionary. For example:
```py
import pyvista

sphere = pyvista.Sphere()
sphere.point_data['arr0'] = range(sphere.n_points)

# show that arr0 is in sphere.point_data
assert 'arr0' in sphere.point_data

# remote arr0
del sphere.point_data['arr0']

# show that arr0 is not in sphere.point_data
assert 'arr0' not in sphere.point_data
```

**Reply by MatthewFlamm** [reply]
I'm noticing that I can't find the documentation for `DataSetAttributes`.  This might be documented in some form but it isn't being included.

**Reply by fgallardo-mpie** [reply]
Hello,

many thanks for your answer =). Maybe this example you posted can be included in the documentation to help other people. I guess it is a frequent feature to use, it is not that uncommon.

Cheers, 
Fran

**Reply by akaszynski** [reply]
Thanks for pointing that out. I'm addressing this in https://github.com/pyvista/pyvista/pull/2109.
